src/huggingface_hub/commands/upload.py

```diff
@@ -155,7 +155,11 @@ def __init__(self, args: Namespace) -> None:
         repo_name: str = args.repo_id.split("/")[-1]  # e.g. "Wauplin/my-cool-model" => "my-cool-model"
         self.local_path: str
         self.path_in_repo: str
-        if args.local_path is None and os.path.isfile(repo_name):
+
+        if "*" in args.local_path:
+            self.local_path = "."
+            self.include = ["*.safetensors"]
+        elif args.local_path is None and os.path.isfile(repo_name):
```

Comment:
```suggestion
        if "*" in args.local_path:
            if self.include is not None:
                raise ValueError("Cannot set `--include` when passing a `local_path` containing a wildcard.")
            self.local_path = "."
            self.include = [args.local_path]

        if args.local_path is None and os.path.isfile(repo_name):
```

Few things I've noticed:
1. we should raise an error if `local_path` has a wildcard and `--include` is set => otherwise we would be silently ignoring the `--include` parameter
2. `self.include` must be set to the local_path provided, `*.safetensors` in the issue description was simply an example
3. we shouldn't use a `elif` after that => the conditions tested after are still relevant. This is why the tests are currently failing.

Reply:
some suggestions in addition to @Wauplin's comment: 
- Maybe we should consider the `?` glob pattern wildcard as well?
- Correct me if i'm wrong, but if a user runs `huggingface-cli upload repo-id ./subfolder/*.safetensors`, setting `local_path = "."` would look in the wrong directory.
- _Nice to have_: Use `glob.glob()` to verify that files actually match the pattern and provide a helpful error if none do.